src/main.rs

```diff
@@ -130,16 +187,44 @@ fn optimize(matches: clap::ArgMatches) {
         }
     };
 
-    let rules = rules_from_str(split_rules);
+    // Get multi-pattern rules. learned_rules are the learned rules from TASO, pre_defined_multi are the hand-specified rules from TASO
+    let multi_patterns = match matches.value_of("multi_rules") {
+        Some(rule_file) => {
+            let learned_rules =
+                read_to_string(rule_file).expect("Something went wrong reading the rule file");
+            let pre_defined_multi = PRE_DEFINED_MULTI.iter().map(|&x| x);
+            let multi_rules: Vec<&str> =
+                learned_rules.split("\n").chain(pre_defined_multi).collect();
+            MultiPatterns::with_rules(multi_rules)
+        }
+        None => {
+            let multi_rules: Vec<&str> = PRE_DEFINED_MULTI.iter().map(|&x| x).collect();
+            MultiPatterns::with_rules(multi_rules)
+        }
+    };
 
     // Run saturation
-    let time_limit_sec = Duration::new(10, 0);
-    let iter_limit = 10;
+    let time_limit_sec = match matches.value_of("n_sec") {
+        Some(s) => Duration::new(s.parse::<u64>().unwrap(), 0),
+        None => Duration::new(10, 0),
```

Comment:
Use `map_or` when the none maps to a default value.

```
let n_sec = matches.value_of("n_sec").map_or(10, |s| s.parse::<u64>().unwrap());
let time_limit_set = Duration::new(n_sec, 0)
```

Reply:
Done